flagsmith-jira-app/src/backend/flagsmith.ts

```diff
@@ -157,22 +159,59 @@ export const readEnvironments: ReadEnvironments = async ({ projectId }) => {
 };
 
 export type ReadFeatures = (args: {
-  projectId?: string;
+  projectIds?: string[];
   environmentId?: string;
 }) => Promise<Feature[]>;
 
 /** Read Flagsmith Features for stored API Key, given Project ID and optional Environment ID */
-export const readFeatures: ReadFeatures = async ({ projectId, environmentId }) => {
+export const readFeatures: ReadFeatures = async ({ projectIds, environmentId }) => {
   const apiKey = await readApiKey();
   checkApiKey(apiKey);
-  if (!projectId) throw new ApiError("Flagsmith project not connected", 400);
-  const params = new URLSearchParams({ is_archived: "false" });
-  if (environmentId) params.set("environment", environmentId);
-  const path = route`/projects/${projectId}/features/?${params}`;
-  const data = (await flagsmithApi(apiKey, path)) as PaginatedModels<Feature>;
-  const results = await unpaginate(apiKey, data);
-  if (results.length === 0) throw new ApiError("Flagsmith project has no features", 404);
-  return results;
+
+  if (!projectIds || projectIds.length === 0) {
+    throw new ApiError("Flagsmith project not connected", 400);
+  }
+
+  const allFeatures: Feature[] = [];
+  const organisationId = await readOrganisationId();
+  const projects = await readProjects({ organisationId });
```

Comment:
I'm a bit concerned here by the amount of requests we are doing.

Basically [here](https://github.com/Flagsmith/flagsmith-jira-integration/pull/28/files#diff-bf954d9e080794b6542189b168bb844f17975aefac90c4163001ac6e42d84513R91) we map over the environments

```
          return await Promise.all(
            environments.map((environment) =>
              readFeatures({ projectIds, environmentId: String(environment.id) }),
```

And making this request for each of them to later do a find.

The projects shouldn't change from one request to another so could we find a way to request once and re-use the result directly ? Maybe having the projects as an argument directly ?